mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs

```diff
@@ -853,10 +855,94 @@ class PacketReader {
+					lenFixedSize = ReadInt ();
+				}
+				switch (etype) {
+					case ElementType.I1: {
+						var val = new sbyte[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+							val[i] = (sbyte)ReadInt ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.U1: {
+						var val = new byte[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+								val[i] = (byte)ReadInt ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.Boolean: {
+						var val = new bool[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+								val[i] = (bool) (ReadInt () != 0);
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.I2: {
+						var val = new short[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+								val[i] = (short)ReadInt ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.U2: {
+						var val = new ushort[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+							val[i] = (ushort)ReadInt ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.Char: {
+						var val = new char[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+							val[i] = (char)ReadInt ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.I4: {
+						var val = new int[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+							val[i] = ReadInt ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.U4: {
+						var val = new uint[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+								val[i] = (uint)ReadInt ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.I8: {
+						var val = new long[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+							val[i] = ReadLong ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.U8: {
+						var val = new ulong[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+							val[i] = (ulong) ReadLong ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.R4: {
+						var val = new float[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+							val[i] = ReadFloat ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+					case ElementType.R8: {
+						var val = new double[lenFixedSize];
+						for (int i = 0; i < lenFixedSize; i++)
+								val[i] = ReadDouble ();
+						return new ValueImpl { Type = etype, Value = val };
+					}
+				}
+				throw new NotImplementedException ("Unable to handle type " + etype);
+			}
 
+			public ValueImpl ReadValue () {
+				ElementType etype = (ElementType)ReadByte ();
 				switch (etype) {
+				case (ElementType)ValueTypeId.VALUE_TYPE_ID_FIXED_ARRAY:
```

Comment:
I would put this case at the end along with the other ValueTypeId cases.

Reply:
okay!